lib/registrations/registration_checker.rb

```diff
@@ -184,6 +185,19 @@ def validate_update_status!(new_status, competition, current_user, target_user,
         raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::ALREADY_REGISTERED_IN_SERIES) if
           new_status == Registrations::Helper::STATUS_ACCEPTED && existing_registration_in_series?(competition, target_user)
 
+        if new_status == Registrations::Helper::STATUS_ACCEPTED && competition.competitor_limit_enabled?
+          raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED) if
+            competition.registrations.competing_status_accepted.count >= competition.competitor_limit
+
+          if !target_user.newcomer? && competition.newcomer_reserved_spots > 0
```

Comment:
In line with my other comment above, this check would make slightly more sense if you use `null` instead of a zero value.

Reply:
See comment above

Reply:
Now that other PR is merged, so you can rethink `null` support